mobius/src/integer_arcs.rs

```diff
@@ -0,0 +1,171 @@
+use std::f64::consts::{PI, TAU};
+
+use thiserror::Error;
+
+use crate::{
+    geometry::{ArcAngles, ArcAnglesError, ArcDirection, Circle, CircularArc, GeneralizedCircle},
+    orthogonal_arcs::{compute_orthogonal_arc, compute_orthogonal_circle},
+};
+
+#[derive(Debug, Error)]
+pub enum IntegerArcError {
+    #[error("{0}")]
+    BadAngles(#[from] ArcAnglesError),
+    #[error("a and b must be distinct: {0}")]
+    DuplicateInt(i64),
+    #[error("n must be nonzero")]
+    ZeroPoints,
+    #[error("out of range value: {0} = {0}")]
+    ValueOutOfRange(String, i64),
+}
+
+/// compute a circle with diameter between a and b on the real line
+pub fn integer_circle(a: i64, b: i64) -> Result<Circle, IntegerArcError> {
```

Comment:
maybe the naming could look like this:

- `integer_arcs::circle_on_line()`
- `integer_arcs::arc_on_line_by_xxx()`
- `integer_arcs::circle_on_circle()`
- `integer_arcs::arc_on_circle_by_xxx()`

To show that we're creating a new circle/arc attached to either the real line or unit circle.